dxp/liferay_global_event_handler.md

```diff
@@ -0,0 +1,213 @@
+
+### First Considerations
+
+First, just consider using the global event handler if necessary. Try to use the default event listener/[callback pattern(if React)](https://reactjs.org/docs/lifting-state-up.html#lifting-state-up) when working. Just use the global event handler when trying to communicate between widgets or not accessible contexts(e.g iframes).
+
+### List of Available Functions
+
+Here is a list of available functions from Liferay’s global variable:
+
+`Liferay.on`
+`Liferay.once`
+`Liferay.onceAfter`
+`Liferay.after`
+`Liferay.before`
+`Liferay.detach`
+`Liferay.detachAll`
+`Liferay.publish`
+`Liferay.fire`
+
+### `Liferay.on`
+
+Description from [yui3/event-target.js at master · liferay/yui3 · GitHub](https://github.com/liferay/yui3/blob/master/src/event-custom/js/event-target.js#L217):
+
+This function subscribes a callback function to a custom event fired by this object or from an object that bubbles its events to this object.
+
+Callback functions for events published with `emitFacade = true` will receive an `EventFacade` as the first argument (typically named “e”). These callbacks can then call `e.preventDefault()` to disable the behaviour published to that event’s `defaultFn` . See the `EventFacade` API for all available properties and methods. Subscribers to non- `emitFacade` events will receive the arguments passed to `fire()` after the event name.
+
+To subscribe to multiple events at once, pass an object as the first argument, where the key:value pairs correspond to the eventName:callback, or pass an array of event names as the first argument to subscribe to all listed events with the same callback.
+
+Returning `false` from a callback is supported as an alternative to calling `e.preventDefault(); e.stopPropagation();` . However, it is recommended to use the event methods whenever possible.
+
+##### Code example:
+
+```js
+var onDestroyPortlet = function () {
+	Liferay.detach('messagePosted', onMessagePosted);
+	Liferay.detach('destroyPortlet', onDestroyPortlet);
+};
+
+Liferay.on('destroyPortlet', onDestroyPortlet);
+```
+
+### `Liferay.once`:
+
+Description from [yui3/event-target.js at master · liferay/yui3 · GitHub](https://github.com/liferay/yui3/blob/master/src/event-custom/js/event-target.js#L136):
+
+Listen to a custom event hosted by the global event handler one time.
+
+This is the equivalent to `on` function except the listener is immediately detached when it is executed.
+
+##### Code Example:
+
+```js
+Liferay.once(this.ns('formReady'), (event) => {
+	const form = Liferay.Form.get(event.formName);
+
+	const field = form.formValidator.getField(this.ns('emailAddress'));
+
+	if (field) {
+		if (this.viewValidDomainsURL) {
+			this.addFieldMessage_(field);
+		}
+	}
```

Comment:
This example is super interesting if you want to learn about `Liferay.Form`, but irrelevant for learning what `Liferay.once` does. I think you could just blow it all away:

```js
Liferay.once('someEvent', (event) => {
    // Called only once, no need to call `detach()`.
});
```

Reply:
Agreed, no need to show unrelated code 😉